packages/tx/src/transaction.ts

```diff
@@ -69,9 +69,9 @@ export default class Transaction {
         to: to && to.length > 0 ? new Address(to) : undefined,
         value: new BN(value),
         data: data ?? emptyBuffer,
-        v: !v?.equals(emptyBuffer) ? new BN(v) : undefined,
-        r: !r?.equals(emptyBuffer) ? new BN(r) : undefined,
-        s: !s?.equals(emptyBuffer) ? new BN(s) : undefined,
+        v: v && !v.equals(emptyBuffer) ? new BN(v) : undefined,
```

Comment:
This won't work if `v` is 0. This is problematic in EIP-2930.

Reply:
Although I am not sure if there exists legacy transactions with a `v` value of 0... They do exist on EIP-2930 transactions, so it can also be "fixed" there, but I am generally more in favor to be correct here; if one inputs any value (also for `r` and `s`) which is 0, then it should create a `BN(0)`.

Reply:
ah hm okay, then maybe I shouldn't remove the validation for `v` to be zero in this commit? https://github.com/ethereumjs/ethereumjs-monorepo/pull/1077/commits/14581b0296745e31b75bcadbe7b9232d17d80ac6

I think I'd prefer that tx v3 continues to support legacy transactions only, and then v4 can support EIP-2930. These accidental `BN(0)` bugs are not fun and I don't think v,r,s should ever be initialized with `BN(0)`, only `undefined` for unsigned transactions, otherwise we need to add more checks around the library (like in `isSigned`). What do you think?

Reply:
> Although I am not sure if there exists legacy transactions with a v value of 0... They do exist on EIP-2930

Are you maybe mixing this up with this v - 27 thing (never understand that completely), I can't imagine that EIP-2930 is changing on the signature format?

Maybe @alcuadrado can have a short look here as well